solvers/optimization_utils.cc

```diff
@@ -170,5 +170,29 @@ int CountConstraintRows(const MathematicalProgram& prog) {
   return n;
 }
 
+double EvalCostGivenSolution(
+    const drake::solvers::MathematicalProgramResult& result,
```

Comment:
I think I mostly understand what is goin on here, but since it is a change in the dairlib/drake stuff I would like to fully understand this